x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/ip_type.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+
+import {
+  StoreParameter,
+  IndexParameter,
+  DocValuesParameter,
+  BoostParameter,
+  NullValueParameter,
+} from '../../field_parameters';
+import { EditFieldSection, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'boost': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'null_value': {
+      return field.null_value !== undefined && field.null_value !== '';
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const IpType = ({ field }: Props) => {
+  return (
+    <>
+      <EditFieldSection>
+        <StoreParameter />
+        <IndexParameter hasIndexOptions={false} />
+        <DocValuesParameter />
+      </EditFieldSection>
+
+      <AdvancedSettingsWrapper>
+        <EditFieldSection>
+          {/* boost */}
+          <BoostParameter defaultToggleValue={getDefaultValueToggle('boost', field.source)} />
+
+          {/* null_value */}
+          <NullValueParameter
```

Comment:
nit: if we have time we could add a custom text field with an "ip" client side validation on it.